mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -841,6 +849,25 @@ impl TakerSwap {
         }
     }
 
+    async fn get_taker_fee_data(&self) -> Result<SwapTxDataMsg, MmError<PaymentInstructionsErr>> {
+        // If taker fee is a lightning payment the payment hash will be sent in the message
+        let taker_fee_data = self.r().taker_fee.as_ref().unwrap().tx_hex_or_hash().0;
```

Comment:
Is unwrapping the option panic safe for this function?

Reply:
Yes, since `taker_fee` is set when the previous event is applied https://github.com/KomodoPlatform/atomicDEX-API/blob/d2f8158f80ee573d8766c92d107d6aef5d9a3eb9/mm2src/mm2_main/src/lp_swap/taker_swap.rs#L720
unwraps are used throughout swap methods this way

Reply:
Then at least lets use `except` with a message like "taker_fee can't must be settled to use `get_taker_fee_data`"